src/hardware/setup.rs

```diff
@@ -23,11 +23,20 @@ const NUM_TCP_SOCKETS: usize = 4;
 const NUM_UDP_SOCKETS: usize = 1;
 const NUM_SOCKETS: usize = NUM_UDP_SOCKETS + NUM_TCP_SOCKETS;
 
+const IPADDR: [u8; 4] = [192, 168, 137, 17];
+
+// Omit DHCP socket, if static IP is used
+#[cfg(feature = "static_ip")]
+const NUM_DHCP_SOCKETS: usize = 0;
+#[cfg(not(feature = "static_ip"))]
+const NUM_DHCP_SOCKETS: usize = 1;
+
 pub struct NetStorage {
     pub ip_addrs: [smoltcp::wire::IpCidr; 1],
 
     // Note: There is an additional socket set item required for the DHCP socket.
-    pub sockets: [smoltcp::iface::SocketStorage<'static>; NUM_SOCKETS + 1],
+    pub sockets: [smoltcp::iface::SocketStorage<'static>;
```

Comment:
I'd just revert these changes. If we have 1 extra storage slot that we don't end up using, it's not a problem (minor memory overhead), but we keep the code a lot cleaner.